### PR TITLE
Add facility_location to FacilityClaim to allow for prioritizing claim address/location in facility details

### PR DESCRIPTION
**Pull request by jwalgran**
## Overview

Add facility_location to FacilityClaim to allow for prioritizing claim address/location in facility details

Connects #1754

## Demo

Screen Shot 2022-08-03 at 9 55 41 AM
Screen Shot 2022-08-03 at 10 18 40 AM

## Testing Instructions

On `develop`:
* As c2@example.com, upload and process [TestGeoCoding.csv](https://github.com/open-apparel-registry/open-apparel-registry/files/9252095/TestGeoCoding.csv). Four of the facilities have real addresses and claim addresses. One (marked 'fake') has a very incorrect claim address, 
* Claim each facility. 
* As c1@example.com, approve each facility.
* As c2@example.com, assign an address to each facility and save. Use the addresses listed as `claim_address` in TestGeoCoding.csv. One address is left intentionally blank to ensure migrating claims without addresses will succeed. 

On this branch:
* Run `./scripts/manage migrate`
* Run `./scripts/manage set_facility_locations_from_claims`
     * You should see one failed facility. The other facilities should succeed. The failure should not prevent the command from completing. 
* Using the 'My Facilities' page, visit each claimed facility in turn. 
     * The claim address and claim location (under 'GPS') should be listed and prioritized, except for Test 5 (fake) and Test 2, where the original facility location should be listed instead. 
     * If the claim location and original facility location are significantly different, you should see both listed; otherwise, only the claim location will be listed. 
     * If the claim address and the original facility address are different, you should see both listed; otherwise, only the claim address will be listed.
* Go to the claim update page for Test 2. Add '990 Spring Garden St, Philadelphia, PA' as the address and save. You should see a failure message. Add a valid address in China and save; it should succeed. 
     * The facility details for Test 2 should now show the new claim location. 
* Clear the address from Test 2 in the claim update page, and save. It should succeed.
     * The facility details for Test 2 should now show the original claim location. 

## Checklist

- [x] `fixup!` commits have been squashed
- [x] CI passes after rebase
- [x] CHANGELOG.md updated with summary of features or fixes, following [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines
- [x] This PR is targeted at the correct branch (`develop` vs. `ogr/develop`)
- [ ] If this PR applies to both OAR and OGR a companion PR has been created

### CODE REVIEW

**Comment by maurizi** [comment]
> The claim address and claim location (under 'GPS') should be listed and prioritized, except for Test 5 (fake) and Test 2, where the original facility location should be listed instead.

This was not the case for Test 5 - it only showed the GPS for the original location, but the claim address was still prioritized even though it was unable to be geocoded. Is that expected behavior?

**Review by maurizi — COMMENTED** [review]
This generally looks good and (mostly) tested as expected, though note my comment above about Test 5 from the provided CSV - not sure if that's expected behavior or not.

File: src/app/src/components/ClaimedFacilitiesDetails.jsx [hunk 1]
```diff
@@ -378,6 +427,11 @@ function ClaimedFacilitiesDetails({
 
     const countryOptions = createCountrySelectOptions(data.countries);
 
+    // const handleAddressChange = e => {
+    //     geocodeAddress(getValueFromEvent(e));
+    //     updateFacilityAddress(e);
+    // };
+
```
[the remark this answers is not in the document]
Comment (TaiWilkin) [reply src/app/src/components/ClaimedFacilitiesDetails.jsx]: Updated in a131e0a033d4f1ad333028aad2fb05658ebc569c

File: src/django/api/migrations/0099_add_facility_location_to_facilityclaim.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Generated by Django 3.2.4 on 2022-07-29 22:26
+
+import django.contrib.gis.db.models.fields
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('api', '0098_add_event'),
```
[the remark this answers is not in the document]
Comment (TaiWilkin) [reply src/django/api/migrations/0099_add_facility_location_to_facilityclaim.py]: I rebased and updated to handle the additional migration

File: src/django/api/views.py [hunk 3]
```diff
@@ -3442,6 +3449,29 @@ def get_claimed_details(self, request, pk=None):
 
             create_extendedfields_for_claim(claim)
 
+            # Conditionally update the facility location if it was changed on
+            # the approved claim. If the location was removed from the claim we
+            # revert the location.
+            latest = claim.history.latest()
+            previous = latest.prev_record
+            prev_location = previous.instance.facility_location
```
[the remark this answers is not in the document]
Comment (TaiWilkin) [reply src/django/api/views.py]: Good call. I was able to update to not use history objects following your suggestion - see ecfc9c247781921ed35bfb97388410d2e771d8d9

**Comment by TaiWilkin** [comment]
@maurizi Thanks for the feedback! After discussion with Justin, I updated the handling of claim addresses without locations (see [d596823](https://github.com/open-apparel-registry/open-apparel-registry/pull/2012/commits/d59682346b098c7e64997f858a060910637faa25)). We now display a warning by the GPS coordinates in this situation. 
Screen Shot 2022-08-05 at 12 15 57 PM

**Review by maurizi — COMMENTED** [review]
looks great 👍

**Review by maurizi — APPROVED** [review]

**Comment by TaiWilkin** [comment]
Thank you for reviewing!

**Pull request merged**